### ISSUE TITLE
ServerlessError: An error occurred: mybucket - mybucketname already exists.

### ISSUE THREAD

**Issue by anloh88** [matt-filion/serverless-external-s3-event#70]
Hello,

after installing the plugin the error "bucket already exists" is still existing. Any ideas?

############################################################################
# serverless.yml

service: my-LambdaFunc

provider:
  name: aws
  runtime: python3.6
  region: eu-central-1

  iamRoleStatements:
    - Effect: "Allow"
      Action:
        - "s3:*"
      Resource: "*"
    - Effect: "Allow"
      Action:
        - "lambda:*"
      Resource: "*"
    - Effect: "Allow"
      Action:
        - "s3:PutBucketNotification"
      Resource:
        Fn::Join:
          - ""
          - - "arn:aws:s3:::*"

plugins:
  - serverless-plugin-existing-s3-cn

functions:
  someFunction:
    handler: handler.hello
    events:
      - existingS3:
          bucket: myfoobucket
          events:
            - s3:ObjectCreated:*

resources:
  Resources:
    mybucket:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: myfoobucket
        NotificationConfiguration:
          LambdaConfigurations:
            - Event: s3:ObjectCreated:*
              Function: arn:aws:lambda:eu-central-1:012345678901:function:my-Lambda-dev-hello:1

############################################################################
sls deploy --aws-profile default

...

Serverless: Operation failed!

  Serverless Error ---------------------------------------

  An error occurred: mybucket - myfoobucket already exists.

  Stack Trace --------------------------------------------

ServerlessError: An error occurred: mybucket - myfoobucket already exists.
    at provider.request.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\lib\monitorStack.js:112:33)
From previous event:
    at AwsDeploy.monitorStack (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\lib\monitorStack.js:26:12)
    at provider.request.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\lib\updateStack.js:95:30)
From previous event:
    at AwsDeploy.update (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\lib\updateStack.js:95:8)
From previous event:
    at AwsDeploy.BbPromise.bind.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\lib\updateStack.js:112:12)
From previous event:
    at AwsDeploy.updateStack (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\lib\updateStack.js:106:8)
From previous event:
    at AwsDeploy.BbPromise.bind.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:129:39)
From previous event:
    at Object.aws:deploy:deploy:updateStack [as hook] (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:125:10)
    at BbPromise.reduce (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:391:55)
From previous event:
    at PluginManager.invoke (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:391:22)
    at PluginManager.spawn (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:409:17)
    at AwsDeploy.BbPromise.bind.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:95:48)
From previous event:
    at Object.deploy:deploy [as hook] (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\deploy\index.js:91:10)
    at BbPromise.reduce (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:391:55)
From previous event:
    at PluginManager.invoke (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:391:22)
    at PluginManager.run (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\classes\PluginManager.js:422:17)
    at variables.populateService.then.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:157:33)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
    at process.topLevelDomainCallback (domain.js:121:23)
From previous event:
    at Serverless.run (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:144:8)
    at serverless.init.then (C:\Users\myuserid\AppData\Roaming\npm\node_modules\serverless\bin\serverless:44:28)

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information -----------------------------
     OS:                     win32
     Node Version:           10.5.0
     Serverless Version:     1.33.2

**Comment by Viktor-K** [comment]
I have the same issue, every deploy same error `"bucket already exists"`.

**Comment by garywoodfine** [comment]
I am getting the same issue!

The annoying part, it seemed to have worked earlier on. Although the Lambda did not respond to the s3:ObjectCreated:*  event.  So I attempted to add some additional login and redeploy.

On re-deploy I encountered the already exists issue.

In attempt to resolve this I completely removed both stacks  i.e. Lambda that creates S3 Bucket and the one configured with the plugin.
  
I attempted to redeploy both. However, the one with plugin only responds with Bucket already exists issue.
I can't seem to get back to getting it working
